finesse/hardware/tc4820.py

```diff
@@ -0,0 +1,212 @@
+"""This module provides an interface to TC4820 temperature controllers.
+
+Decimal numbers are used for values sent to and read from the device as the values are
+base-10 and using floats could cause rounding errors.
+"""
+import logging
+from decimal import Decimal
+from typing import Any
+
+from serial import Serial, SerialException
+
+
+class MalformedMessageError(Exception):
+    """Raised when a message sent or received was malformed."""
+
+
+class TC4820:
+    """An interface for TC4820 temperature controllers."""
+
+    MAX_POWER = 511
+    """The maximum value for the power property."""
+
+    def __init__(self, serial: Serial, max_attempts: int = 3) -> None:
+        """Create a new TC4820 from an existing serial device.
+
+        Args:
+            serial: Serial device
+            max_attempts: Maximum number of attempts for requests
+        """
+        if max_attempts < 1:
+            raise ValueError("max_attempts must be at least 1")
+
+        self.serial = serial
+        self.max_attempts = max_attempts
+
+    @staticmethod
+    def create(
+        port: str,
+        baudrate: int = 115200,
+        timeout: float = 1.0,
+        max_attempts: int = 3,
+        *serial_args: Any,
+        **serial_kwargs: Any,
+    ) -> "TC4820":
+        """Create a new TC4820.
+
+        Args:
+            port: Serial port name
+            baudrate: Serial port baudrate
+            timeout: How long to wait for read/write operation
+            max_attempts: Maximum number of attempts for requests
+            serial_args: Extra arguments to Serial constructor
+            serial_kwargs: Extra keyword arguments to Serial constructor
+        """
+        # If the user hasn't specified an explicit timeout for write operations, then
+        # use the same as for read operations
+        if "write_timeout" not in serial_kwargs:
+            serial_kwargs["write_timeout"] = timeout
```

Comment:
I think we have had this conversation before, but I would put the description of what the method is doing in the docstring. Especially this one, which is not an implementation detail but something that the user should know.

Reply:
Good point